builtin/common/misc_helpers.lua

```diff
@@ -498,6 +498,27 @@ function core.explode_scrollbar_event(evt)
 end
 
 --------------------------------------------------------------------------------
+local function format(n)
+	local str = string.format("%X", n)
+		if str:len() == 1 then
```

Comment:
Use %02X to zero pad the hexadecimal number. Also get rid of the format function, as it is not needed once you do this.

You may also be able to do `string.format("%02X%02X%02X", red, green, blue)` but I'm not certain Lua supports that.